lib/utils.pm

```diff
@@ -761,7 +761,8 @@ sub handle_login {
             select_user_gnome($myuser);
         }
     }
-    assert_screen 'displaymanager-password-prompt', no_wait => 1;
+    record_soft_failure('bsc#1122664 - password prompt is not focused') if (check_screen 'displaymanager-unfocused-password-prompt');
```

Comment:
and again this can not work reliably because `check_screen` has a 0 timeout, meaning there is no waiting between the former selection of user and the password prompt.
Please use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen` with non-zero timeout to prevent introducing any timing dependant behaviour, to save test execution time as well as state more explicitly from the testers point of view what are the expected alternatives. For example:

```perl
assert_screen([qw(yast2_console-finished yast2_missing_package)]);
if (match_has_tag('yast2_missing_package')) {
    send_key 'alt-o';  # confirm package installation
    assert_screen 'yast2_console-finished';
}
```

Reply:
With a blinking cursor, using multi-tag assertion you have 50% of chances of getting false positives. But, anyway I just realized that asserting an unfocused textbox is also wrong. Because the unfocused textbox looks the same as a focused textbox on an instance when the cursor is off.

Reply:
1. There is the option "no_wait" on e.g. `assert_screen` to cover "blinking cursor" if you must
2. Better avoid the blinking cursor, e.g. with a proper needle design, could be an exclude area
3. Are you *sure* unfocussed looks the same as focussed but cursor off? If you mean by "cursor off", while it's blinking I remember in gdm that there is still a small white border around the box

Reply:
1. assert_screen will break the test on unfocused textbox
2. We are exactly trying to match this area, we cannot exclude it.
3. Yes, see: https://progress.opensuse.org/issues/46223#note-6